server/src/domain/media/media.service.ts

```diff
@@ -59,51 +58,89 @@ export class MediaService {
 
-    await this.assetRepository.save({ id: asset.id, resizePath: jpegThumbnailPath });
-
+    const resizePath = this.getPath(asset, 'jpeg');
+    await this.assetRepository.save({ id: asset.id, resizePath });
     return true;
   }
 
+  async generateImageThumbnail(asset: AssetEntity, format: 'jpeg' | 'webp') {
+    const { thumbnail } = await this.configCore.getConfig();
+    const size = format === 'jpeg' ? thumbnail.jpegSize : thumbnail.webpSize;
+    const thumbnailOptions = { format, size, wideGamut: thumbnail.wideGamut, quality: thumbnail.quality };
+    const thumbnailPath = this.getPath(asset, format);
+    await this.mediaRepository.resize(asset.originalPath, thumbnailPath, thumbnailOptions);
+  }
+
+  async generateVideoThumbnail(asset: AssetEntity, format: 'jpeg' | 'webp') {
+    const { ffmpeg, thumbnail } = await this.configCore.getConfig();
+    const size = format === 'jpeg' ? thumbnail.jpegSize : thumbnail.webpSize;
+    const { videoStreams } = await this.mediaRepository.probe(asset.originalPath);
+    const mainVideoStream = this.getMainVideoStream(videoStreams);
+    if (!mainVideoStream) {
+      throw new UnsupportedMediaTypeException(
+        `Could not extract thumbnail for asset ${asset.id}: no video streams found`,
+      );
+    }
+    const thumbnailPath = this.getPath(asset, format);
+    const config = { ...ffmpeg, targetResolution: size.toString() };
+    const options = new ThumbnailConfig(config).getOptions(mainVideoStream);
+    await this.mediaRepository.transcode(asset.originalPath, thumbnailPath, options);
+  }
+
   async handleGenerateWebpThumbnail({ id }: IEntityJob) {
     const [asset] = await this.assetRepository.getByIds([id]);
     if (!asset || !asset.resizePath) {
       return false;
     }
 
-    const webpPath = asset.resizePath.replace('jpeg', 'webp').replace('jpg', 'webp');
+    switch (asset.type) {
+      case AssetType.IMAGE:
+        try {
+          await this.generateImageThumbnail(asset, 'webp');
+          this.logger.log(`Successfully generated WebP image thumbnail ${asset.id}`);
+        } catch (err) {
+          this.logger.error(`Could not generate WebP image thumbnail for asset ${asset.id}: ${err}`);
+          return false;
+        }
+        break;
+      case AssetType.VIDEO:
+        try {
+          await this.generateVideoThumbnail(asset, 'webp');
+          this.logger.log(`Successfully generated WebP video thumbnail ${asset.id}`);
+        } catch (err) {
+          this.logger.error(`Could not generate WebP video thumbnail for asset ${asset.id}: ${err}`);
+          return false;
+        }
+        break;
+      default:
+        throw new UnsupportedMediaTypeException(`Unsupported asset type for thumbnail generation: ${asset.type}`);
+    }
 
-    const { thumbnail } = await this.configCore.getConfig();
-    await this.mediaRepository.resize(asset.resizePath, webpPath, { size: thumbnail.webpSize, format: 'webp' });
+    const webpPath = this.getPath(asset, 'webp');
```

Comment:
nit: it's pretty clear from context this will be webp so it could be 'path' or 'thumbnailPath' if you really want to be explicit.